src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13619,7 +13627,7 @@ private new struct ResetPoint
 
         private SyntaxToken MergeTokens(SyntaxToken s1, SyntaxToken s2, SyntaxKind kind)
         {
-            if (s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0)
+            if (!s1.ContainsDiagnostics && !s2.ContainsDiagnostics && s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0)
```

Comment:
oh interesting.  i don't love this approach.  i think we should instead always just merge, and say the caller only calls into thsi when they have both of  these tokens.